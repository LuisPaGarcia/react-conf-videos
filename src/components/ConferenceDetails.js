import React from 'react'

const ConferenceDetails = ({ title, date }) => <h1> { title + ' ' + date } </h1>

export default ConferenceDetails
