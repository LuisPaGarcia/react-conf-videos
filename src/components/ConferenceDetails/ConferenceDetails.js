import React from 'react'
import styles from './ConferenceDetails.scss'

const ConferenceDetails = ({ title, date }) => (
  <section className={styles.root}>
    <h1> { title + ' ' + date } </h1>
  </section>
)

export default ConferenceDetails
