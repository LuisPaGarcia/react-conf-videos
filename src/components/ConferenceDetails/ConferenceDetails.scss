.root {
  width: 100%;
  background: white;
  padding: 1rem 2rem;
  box-sizing: border-box;
  
  h2.details {
    font-size: 1.6rem;
  }
}