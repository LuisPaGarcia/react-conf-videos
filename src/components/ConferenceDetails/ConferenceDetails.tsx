import * as React from 'react'

import styles from './ConferenceDetails.scss'

type Props = {
  title: string
  date: string
}

const ConferenceDetails: React.SFC<Props> = ({ title, date }) => (
  <section className={styles.root}>
    <h2 className={styles.details}> { title + ' ' + date } </h2>
  </section>
)

export default ConferenceDetails
