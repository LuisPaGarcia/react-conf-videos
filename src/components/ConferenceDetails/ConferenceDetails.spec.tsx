import * as React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import ConferenceDetails from './ConferenceDetails'

describe('ConferenceDetails', () => {
  it('should render and display the title and date', () => {
    // arrange 
    const props = {title: 'Fake Conf Title', date: 'XX/XX/XXXX'}
    const expectedRenderString = `${props.title} ${props.date}`
    // act
    const wrapper = shallow(<ConferenceDetails {...props} />)

    // assert
    expect(toJSON(wrapper)).toMatchSnapshot()
    expect(wrapper.find('h1').text().trim()).toEqual(expectedRenderString)
  })
})