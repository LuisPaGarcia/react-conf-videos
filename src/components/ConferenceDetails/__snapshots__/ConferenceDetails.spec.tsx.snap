// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConferenceDetails should render and display the title and date 1`] = `
<section
  className="root"
>
  <h2
    className="details"
  >
     
    Fake Conf Title XX/XX/XXXX
     
  </h2>
</section>
`;
